packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DeleteObjectTreeItemEventHandler.java

```diff
@@ -55,16 +58,32 @@ public boolean canHandle(IEditingContext editingContext, TreeItem treeItem) {
     }
 
     @Override
-    public IStatus handle(IEditingContext editingContext, TreeItem treeItem) {
-        Optional<Object> optionalObject = this.objectSearchService.getObject(editingContext, treeItem.getId());
-        if (optionalObject.isPresent()) {
-            Object object = optionalObject.get();
-            this.editService.delete(object);
-
-            return new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, Tree tree) {
+        IStatus result = new Failure("");
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
+        if (optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, treeItem.getId());
+            var object = optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager);
+            if (object != null) {
+                this.editService.delete(object);
+                result = new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+            } else {
+                this.logger.warn("The object with the id {} does not exist", treeItem.getId());
+            }
         } else {
-            this.logger.warn("The object with the id {} does not exist", treeItem.getId());
+            this.logger.warn("Unable to retireve the tree description with the id {}", tree.getDescriptionId());
         }
-        return new Failure("");
+        return result;
+    }
+
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done